src/spikeinterface/core/waveform_extractor.py

```diff
@@ -146,7 +147,10 @@ def load_from_folder(cls, folder, with_recording=True, sorting=None):
         folder = Path(folder)
         assert folder.is_dir(), f'This waveform folder does not exists {folder}'
 
-        if not with_recording:
+        if isinstance(with_recording, BaseRecording):
+            recording = with_recording
+            rec_attributes = None
+        elif not with_recording:
```

Comment:
Can you change the docstring of the `extract_waveforms` function to reflect the new behavior?

Reply:
I'm assuming you mean `load_waveforms`. It's done.